controlplane/kubeadm/internal/etcd/backoff_adapter.go

```diff
@@ -0,0 +1,186 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package etcd
+
+import (
+	"context"
+	"time"
+
+	"go.etcd.io/etcd/clientv3"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/klog/klogr"
+)
+
+// Log is the global logger used. Global var can be swapped out if necessary.
+var Log = klogr.New()
```

Comment:
Would it make sense to make `Log` a field of `EtcdBackoffAdapter` rather than a global here?

Reply:
If we need to configure a logger for the etcd response formatter separately from the backoff adapter then yes, we can move it. I'm going to leave it here for now though as we do not need this functionality.

This is also in the internal/ package so we don't have to worry about breaking a public API and can change this on a whim.